File: backend/minecraft/players.go
package minecraft

type PlayerCacheEntry struct {
	Name      string `json:"name"`
	UUID      string `json:"uuid"`
	ExpiresOn string `json:"expiresOn"`
}

type OpEntry struct {
	UUID  string `json:"uuid"`
	Name  string `json:"name"`
	Level int    `json:"level"`
}

type BanEntry struct {
	UUID    string `json:"uuid"`
	Name    string `json:"name"`
	Created string `json:"created"`
	Source  string `json:"source"`
	Expires string `json:"expires"`
	Reason  string `json:"reason"`
}

File: frontend/src/lib/types/players.ts
export interface PlayerCache {
    name: string;
    uuid: string;
    expiresOn: string;
    online: boolean;
}

export interface OpEntry {
    uuid: string;
    name: string;
    level: number;
    bypassesPlayerLimit: boolean;
}

export interface BanEntry {
    uuid: string;
    name: string;
    created: string;
    source: string;
    expires: string;
    reason: string;
}

export interface PlayerActionRequest {
    player: string;
    action: 'op' | 'deop' | 'ban' | 'pardon' | 'kick' | 'whitelist_add' | 'whitelist_remove';
    reason?: string;
}
